pkg/cli/admin/top/persistentvolumeclaims.go

```diff
@@ -0,0 +1,382 @@
+			}
+		}
+	}
+
+	// if more pvc are requested as args but one of them does not not exist
+	if len(args) != 0 && len(promOutput.Data.Result) != len(args) {
+		resultingPvc := make(map[string]bool)
+		for _, _promOutputDataResult := range promOutput.Data.Result {
+			pvcName := _promOutputDataResult.Metric["persistentvolumeclaim"]
+			resultingPvc[pvcName] = true
+		}
+		for _, arg := range args {
+			_, pvcPresentInResult := resultingPvc[arg]
+			if !pvcPresentInResult {
+				return fmt.Errorf("persistentvolumeclaim %q not found in %s namespace.", arg, o.Namespace)
+			}
+		}
+
+	}
+	headers := []string{"NAMESPACE", "NAME", "USAGE(%)"}
+	pvcInfos := []persistentVolumeClaimInfo{}
+	infos := []Info{}
+	for _, _promOutputDataResult := range promOutput.Data.Result {
+		namespaceName := _promOutputDataResult.Metric["namespace"]
+		pvcName := _promOutputDataResult.Metric["persistentvolumeclaim"]
+		usagePercentage := _promOutputDataResult.Value[1]
+		valueFloatLong, _ := strconv.ParseFloat(usagePercentage.(string), 64)
+		valueFloat := fmt.Sprintf("%.2f", valueFloatLong)
+		if len(pvcInfos) > 0 {
+			if !(namespaceName == pvcInfos[len(pvcInfos)-1].Namespace && pvcName == pvcInfos[len(pvcInfos)-1].Name) {
+				pvcInfos = append(pvcInfos, persistentVolumeClaimInfo{Namespace: namespaceName, Name: pvcName, UsagePercentage: valueFloat})
+				infos = append(infos, persistentVolumeClaimInfo{Namespace: namespaceName, Name: pvcName, UsagePercentage: valueFloat})
+			}
+		} else {
+			pvcInfos = append(pvcInfos, persistentVolumeClaimInfo{Namespace: namespaceName, Name: pvcName, UsagePercentage: valueFloat})
+			infos = append(infos, persistentVolumeClaimInfo{Namespace: namespaceName, Name: pvcName, UsagePercentage: valueFloat})
+		}
+	}
+
+	Print(o.Out, headers, infos)
+	return nil
+}
+
+func GetPersistentVolumeClaims(ctx context.Context, getRoute RouteGetter, bearerToken string, namespace string, insecureTLS bool, args []string) ([]byte, error) {
+	uri := &url.URL{
+		Scheme: "https",
+		Path:   "/api/v1/query",
+	}
+	query := ""
+	claimNames := ".*"
+	if namespace != "" {
+		if len(args) > 0 {
+			claimNames = strings.Join(args, "|")
+		}
+		query = fmt.Sprintf(`100*kubelet_volume_stats_used_bytes{persistentvolumeclaim=~"%s", namespace="%s"}/kubelet_volume_stats_capacity_bytes{persistentvolumeclaim=~"%s", namespace="%s"}`, claimNames, namespace, claimNames, namespace)
+	} else {
+		query = `100*kubelet_volume_stats_used_bytes{persistentvolumeclaim=~".*"}/kubelet_volume_stats_capacity_bytes{persistentvolumeclaim=~".*"}`
+	}
+	urlParams := url.Values{}
+	urlParams.Set("query", query)
+	uri.RawQuery = urlParams.Encode()
+
+	persistentVolumeClaimsBytes, err := getWithBearer(ctx, getRoute, "openshift-monitoring", "prometheus-k8s", uri, bearerToken, insecureTLS)
+	if err != nil {
+		klog.V(4).Info(fmt.Errorf("failed to get persistentvolumeclaims from Prometheus: %w", err))
```

Comment:
I agree with you that we should return it instead of hiding it, and it was like this before, then I changed the behaviour to accommodate the request made [here](https://github.com/openshift/enhancements/pull/1704#discussion_r1814842112).
I'll revert the code as it was previously.